mycelo/config/config.go

```diff
@@ -0,0 +1,191 @@
+package config
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"math/big"
+	"os"
+
+	"github.com/ethereum/go-ethereum/internal/fileutils"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+type Environment struct {
```

Comment:
i would try to remove `GenesisConfig` here, since it's only ever used to generate the genesis, and not needed afterwards. 

The env should give you access to:
  * EnvConfig (i would rename to Config)
  * Genesis.json location
 
Paths is kinda mixed.. has the paths to geth which you don't always have, and paths to genesis-config.json which is not always present.